src/snakemake/resources.py

```
@@ -521,84 +444,493 @@
+        "full": {
+            "mem": "min(max(2*input.size_mb, 1000), 8000)",
+            "disk": "max(2*input.size_mb, 1000)",
+            "tmpdir": "system_tmpdir",
+        },
+        "bare": {"tmpdir": "system_tmpdir"},
+    }
 
-    if resources_args is not None:
-        valid = re.compile(r"[a-zA-Z_]\w*$")
+    def __init__(self, mapping: dict[str, Resource] | None = None):
+        if mapping == None:
```

Ruff (0.8.2)

Comparison to `None` should be `cond is None`

Replace with `cond is None`

(E711)